apps/frontend/src/store/search.ts

```diff
@@ -0,0 +1,550 @@
+  }
+
+  @Action
+  removeStatusSearch(status: ExtendedControlStatus) {
+    this.removeStatusFilter(status);
+    // If we still have any status filters
+    if (this.statusFilter.length !== 0) {
+      const newSearch = this.searchTerm.replace(
+        /status:"(.*?)"/gm,
+        `status:"${this.statusFilter.join(',')}"`
+      );
+      this.context.commit('SET_SEARCH', newSearch);
+    } // Otherwise remove the status filter text from the search bar
+    else {
+      const newSearch = this.searchTerm.replace(/status:"(.*?)"/gm, '');
+      this.context.commit('SET_SEARCH', newSearch);
+    }
+    this.parseSearch();
+  }
+
+  @Action
+  addStatusFilter(status: ExtendedControlStatus | ExtendedControlStatus[]) {
+    this.context.commit('ADD_STATUS', status);
+  }
+
+  @Action
+  removeStatusFilter(status: ExtendedControlStatus) {
+    this.context.commit('REMOVE_STATUS', status);
+  }
+
+  @Action
+  clearStatusFilter() {
+    this.context.commit('CLEAR_STATUS');
+  }
+
+  @Action
+  setStatusFilter(status: ExtendedControlStatus[]) {
+    this.context.commit('SET_STATUS', status);
+  }
+
+  /** Adds a status filter */
+  @Mutation
+  ADD_STATUS(status: ExtendedControlStatus | ExtendedControlStatus[]) {
+    this.statusFilter = this.statusFilter.concat(status);
+  }
+
+  /** Removes a status filter */
+  @Mutation
+  REMOVE_STATUS(status: ExtendedControlStatus) {
+    this.statusFilter = this.statusFilter.filter(
+      (filter) => filter.toLowerCase() !== status.toLowerCase()
+    );
+  }
+
+  @Mutation
+  CLEAR_STATUS() {
+    this.statusFilter = [];
+  }
+
+  @Mutation
+  SET_STATUS(status: ExtendedControlStatus[]) {
+    this.statusFilter = status;
+  }
+
+  // Severity filtering
+
+  /** Adds or replaces severity search in the search bar */
+  @Action
+  addSeveritySearch(severity: Severity) {
+    // If we already have search filtering
+    if (this.searchTerm.trim() !== '') {
+      // If our current filters include severity
+      if (this.searchTerm.toLowerCase().indexOf('severity:') !== -1) {
+        const newSearch = this.searchTerm.replace(
+          /severity:"(.*?)"/gm,
+          `severity:"${this.severityFilter.concat(severity).join(',')}"`
+        );
+        this.context.commit('SET_SEARCH', newSearch);
+      } // We have a filter already, but it doesn't include severity
+      else {
+        const newSearch = `${this.searchTerm} severity:"${this.severityFilter
+          .concat(severity)
+          .join(',')}"`;
+        this.context.commit('SET_SEARCH', newSearch);
+      }
+    }
+    // We don't have any search yet
+    else {
+      this.context.commit('SET_SEARCH', `severity:"${severity}"`);
+    }
+    this.parseSearch();
+  }
+
+  @Action
+  removeSeveritySearch(severity: Severity) {
+    this.removeSeverity(severity);
+    if (this.severityFilter.length !== 0) {
+      // If we still have any severity filters
+      const newSearch = this.searchTerm.replace(
+        /severity:"(.*?)"/gm,
```

Comment:
`(.*?)` === `(.*)`, the `*` means "0 or more"

Reply:
The ? here is to only match to the end of the first result, because if there is a field after the one currently being edited it will match to the end as well, e.g [this](https://regex101.com/r/7YoLqj/1) vs [this](https://regex101.com/r/oQawzZ/1)